dae/dae/genomic_resources/reference_genome.py

```diff
@@ -268,10 +501,122 @@ def calc_info_hash(self):
+                f"{self.resource.resource_id}_count_nucleotides_"
+                f"{chrom}_{start}_{end}",
+                ReferenceGenome._do_chrom_statistic,
+                [self.resource, chrom, start, end],
+                []
+            ))
+        chroms = self.chromosomes
+        merge_task = task_graph.create_task(
+            f"{self.resource.resource_id}_merge_chrom_statistics_{chrom}",
+            ReferenceGenome._merge_chrom_statistics,
+            [chroms, *chrom_tasks],
+            chrom_tasks
+        )
+        save_task = task_graph.create_task(
+            f"{self.resource.resource_id}_save_chrom_statistics_{chrom}",
+            ReferenceGenome._save_chrom_statistics,
+            [self.resource, merge_task],
+            [merge_task]
+        )
+
+        return chrom_tasks, merge_task, save_task
+
+    def _add_global_stat_task(self, task_graph):
+        return task_graph.create_task(
+            f"{self.resource.resource_id}_save_chrom_statistics",
+            ReferenceGenome._do_global_statistic,
+            [self.resource],
+            []
+        )
+
+    @staticmethod
+    def _do_chrom_statistic(resource, chrom, start, end):
+        impl = build_reference_genome_from_resource(resource)
+        statistic = ChromosomeStatistic(chrom)
+        with impl.open():
+            for nuc in impl.get_sequence(chrom, start, end):
+                statistic.add_value(nuc)
+
+        statistic.finish()
+        return {
+            chrom: statistic
+        }
+
+    @staticmethod
+    def _merge_chrom_statistics(chroms, *chrom_tasks):
+        res: dict[str, Optional[ChromosomeStatistic]] = {
+            chrom: None for chrom in chroms}
+
+        for chrom_task_result in chrom_tasks:
+            for task_chrom, task_result in chrom_task_result.items():
+                if res[task_chrom] is None:
+                    res[task_chrom] = task_result
+                else:
+                    res[task_chrom].merge(task_result)
+        return res
+
+    @staticmethod
+    def _save_chrom_statistics(resource, merged_statistics):
+        proto = resource.proto
+        for chrom, chrom_statistic in merged_statistics.items():
+            if chrom_statistic is None:
+                logger.warning("Chrom statistic for %s is None", chrom)
+                continue
+            with proto.open_raw_file(
+                resource,
+                f"{ReferenceGenomeStatistics.get_statistics_folder()}"
+                f"/{ReferenceGenomeStatistics.get_chrom_file(chrom)}",
+                mode="wt"
+            ) as outfile:
+                outfile.write(chrom_statistic.serialize())
+        return merged_statistics
+
+    @staticmethod
+    def _do_global_statistic(resource):
+        impl = build_reference_genome_from_resource(resource)
+        with impl.open():
+            statistic = GenomeStatistic(impl.chromosomes)
+            for pair in impl.pair_iter():
```

Comment:
This one task is going to iterate over the whole reference genome. This is going to be painfully slow.

Let us discuss this.